src/numeric/src/integers.cairo

```diff
@@ -0,0 +1,77 @@
+use alexandria_math::BitShift;
+
+#[generate_trait]
+impl U32Impl of U32Trait {
```

Comment:
probably worth converting that trait in a generic trait that can be implemented over any type